### PR TITLE
Add example of c calling rust

### PR DESCRIPTION
**Pull request by mfarrugi**
Addresses #55.

### CODE REVIEW

**Comment by mfarrugi** [comment]
Is the buildkite presubmit intended to not be publicly visible (with an account)?

**Comment by acmcarther** [comment]
I think that's a design shortcoming. The detail page there lands on an administrative page. FWICT, you can see the build results without logging in.

File: examples/ffi/c_calling_rust/BUILD [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    name = "wrapper",
+    srcs = [":rusty"]
+)
+
+cc_test(
+    name = "main",
+    srcs = ["main.c"],
+    deps = [":wrapper"]
+)
+
```
Comment (acmcarther) [inline examples/ffi/c_calling_rust/BUILD]: Nit: Repo style is no extra newline (this file and rest)

File: examples/ffi/c_calling_rust/BUILD [hunk 2]
```diff
@@ -0,0 +1,20 @@
+load("@//rust:rust.bzl", "rust_library", "rust_test", "rust_binary")
```
Comment (acmcarther) [inline examples/ffi/c_calling_rust/BUILD]: Nit: Other examples have a visibility defined:
```
package(default_visibility = ["//visibility:public"])
```

Comment (mfarrugi) [reply examples/ffi/c_calling_rust/BUILD]: Should we weaken the rest of the examples' visibilities?

**Comment by mfarrugi** [comment]
I didn't move the matrix example because I had already made changes to it in #61, I can follow suit in that pr

File: examples/ffi/c_calling_rust/BUILD (hunk 2 above)
Comment (mfarrugi) [reply examples/ffi/c_calling_rust/BUILD]: I'm not excited about anyone depending on this or other examples

**Comment by mfarrugi** [comment]
Updated #61 to include "rust_calling_c" rename as well.

**Comment by mfarrugi** [comment]
@acmcarther Anything else for this one?

**Comment by acmcarther** [comment]
LGTM, merging!

**Pull request merged**